File: src/utils/file-utils.js
import fs from 'node:fs/promises';
import path from 'node:path';

export async function ensureDir(dirPath) {
  await fs.mkdir(dirPath, { recursive: true });
}

export function joinPath(...segments) {
  return path.join(...segments);
}

export async function ensureOutputDirs(pathsConfig) {
  await ensureDir(pathsConfig.screenshotDir);
  await ensureDir(pathsConfig.resultsDir);
  await ensureDir(joinPath(pathsConfig.screenshotDir, 'interactions'));
}

export async function readJsonFile(filePath) {
  const raw = await fs.readFile(filePath, 'utf-8');
  return JSON.parse(raw);
}

export async function writeJsonFile(filePath, data) {
  const json = JSON.stringify(data, null, 2);
  await fs.writeFile(filePath, json, 'utf-8');
}

export function buildTimestampForFileName(date = new Date()) {
  const yyyy = date.getFullYear();
  const mm = String(date.getMonth() + 1).padStart(2, '0');
  const dd = String(date.getDate()).padStart(2, '0');
  const hh = String(date.getHours()).padStart(2, '0');
  const min = String(date.getMinutes()).padStart(2, '0');
  const ss = String(date.getSeconds()).padStart(2, '0');

  return `${yyyy}-${mm}-${dd}_${hh}-${min}-${ss}`;
}

File: src/utils/file_utils.py
import json
import os
from datetime import datetime


def ensure_dir(dir_path):
    os.makedirs(dir_path, exist_ok=True)


def join_path(*segments):
    return os.path.join(*segments)


def ensure_output_dirs(paths_config):
    ensure_dir(paths_config["screenshotDir"])
    ensure_dir(paths_config["resultsDir"])


def read_json_file(file_path):
    with open(file_path, "r", encoding="utf-8") as file:
        return json.load(file)


def write_json_file(file_path, data):
    parent_dir = os.path.dirname(file_path)
    if parent_dir:
        ensure_dir(parent_dir)
    with open(file_path, "w", encoding="utf-8") as file:
        json.dump(data, file, indent=2, ensure_ascii=False)


def write_text_file(file_path, content):
    parent_dir = os.path.dirname(file_path)
    if parent_dir:
        ensure_dir(parent_dir)
    with open(file_path, "w", encoding="utf-8") as file:
        file.write(content)


def build_timestamp_for_file_name(date=None):
    value = date or datetime.now()
    yyyy = value.strftime("%Y")
    mm = value.strftime("%m")
    dd = value.strftime("%d")
    hh = value.strftime("%H")
    minute = value.strftime("%M")
    ss = value.strftime("%S")
    return f"{yyyy}-{mm}-{dd}_{hh}-{minute}-{ss}"
